presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControl.java

```diff
@@ -78,15 +85,22 @@ public SystemAccessControl create(Map<String, String> config)
             checkState(
                     configFileName != null,
                     "Security configuration must contain the '%s' property", CONFIG_FILE_NAME);
+            if (config.containsKey(PERIOD)) {
+                return refreshingSystemAccessControl(
+                        () -> {
+                            log.info("Refreshing access control, because '%s' was modified", configFileName);
+                            return create(configFileName);
+                        },
+                        wasFileModified(toPath(configFileName)),
+                        Duration.valueOf(config.get(PERIOD)));
```

Comment:
If this fails, it would be good to include "security.refresh-period" somewhere in the exc msg